src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/RollingFileTraceListener.cs

```diff
@@ -403,10 +403,13 @@ public override void Write(char value)
         /// An I/O error occurs.
         /// </exception>
         /// <filterpriority>1</filterpriority>
-        public override void Write(char[] buffer)
+        public override void Write(char[]? buffer)
         {
             base.Write(buffer);
-            Tally += Encoding.GetByteCount(buffer);
+            if (buffer is not null)
```

Comment:
nit: Replacing this with a conditional will make the code more compact, what do you think ?

```
Tally += (buffer is not null) ? Encoding.GetByteCount(buffer) : 0;
```